### PR TITLE
[TV 2 Contributions] More additions to 2020Q1

### PR DESCRIPTION
**Pull request by thomasslee97**
Adds the following properties used by core:
- Transition properties.
  - Allows adlibs to become parts which have transition properties when queued, enabling a full in-and-out transition with autonext to be adlibbed.
- Adds select type to config manifest.
  - Allows the blueprints to expose dropdowns that can be either single-select or multiselect. The blueprints can present a list of acceptable inputs, or they can use the "Source Layer" or "Layer Mapping" types to request that core presents a list of available source layers or layer mappings to the user.
- Table column ranks.
  - Allows the blueprints to change the order in which columns in tables are presented by assigning a rank to each column (lowest rank on the left, highest on the right).

### CODE REVIEW

**Comment by Julusian** [comment]
Note to self: Do the complex interfaces behave well with typescript to json schema converters?

File: src/rundown.ts [hunk 1]
```diff
@@ -246,6 +246,12 @@ export interface IBlueprintPieceGeneric {
 	toBeQueued?: boolean
 	/** Array of items expected to be played out. This is used by playout-devices to preload stuff. */
 	expectedPlayoutItems?: ExpectedPlayoutItemGeneric[]
+	/** When queued, should the adlib autonext */
+	adlibAutoNext?: boolean
+	/** When queued, how much overlap with the next part */
+	adlibAutoNextOverlap?: number
+	/** When queued, block transition at the end of the part */
+	adlibDisableOutTransition?: boolean
```
Comment (jstarpl) [inline src/rundown.ts:254]: The "disable out transition" feature seems to be very specific, does it make sense to implement it, only to port it to AdLib Actions/Functions later, which I think is the more universal solution?

Comment (Julusian) [reply src/rundown.ts:254]: I think these should be added, as it kind of serves a different use case to the adlib-actions (for now at least). these properties will be used when using the sticky hotkeys, which could be relevant for tv2 if they want to recall the last transition

**Comment by Julusian** [comment]
I've just checked, and the `ts-json-schema-generator` package can happily produce valid schema from the complex config types used.
The results look kinda funky, but do work as intended:
```
"multiple": {
  "enum": [
    false
  ],
  "type": "boolean"
},
```

**Pull request merged**